### PR TITLE
Assign placeholder for missing field values in alerts

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
| Related issue |
| --- |
| #25484 |

# Description
This PR introduces a new helper function, `populateField`, which simplifies the process of populating fields in a `nlohmann::json` object with appropriate values based on their types. This function ensures that empty or undefined values are handled consistently, improving the robustness of the alert system.

The field replacement is implemented according to the current index template used by alerts, and a unit test (UT) has been added to cover this new scenario.

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventPackageAlertDetailsBuilder.hpp
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventPackageAlertDetailsBuilder.hpp:1]: As seen here:
- https://github.com/wazuh/wazuh/blob/520e9611657f8da9b28ce6b91b1860d20d98fc82/src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp#L218-L221

Some fields can be empty when we assign them

We should add a more generic logic. Maybe iterate the fields of the final array and fill the empty strings with the corresponding "-" or "-1"

**Review by sebasfalcone — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventPackageAlertDetailsBuilder.hpp
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventPackageAlertDetailsBuilder.hpp:1]: Nice idea, @sebasfalcone! 😊

I’ve created a helper function to handle the logic and applied it to the fields you mentioned. If you'd like me to add it to more fields or move the helper function to a different folder, just let me know!

Thanks for the suggestion! 🦾

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/fieldAlertHelper.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        else if constexpr (std::is_same_v<std::decay_t<T>, double>)
+        {
+            // Return -1.0 if the double value rounds to zero, otherwise return the original value
+            return (Utils::floatToDoubleRound(field, 2) == 0) ? -1 : field;
+        }
+        else
+        {
+            // Throw an exception if an unsupported type is passed to the function
+            throw std::runtime_error("Unsupported type for fillEmptyOrNegative");
+        }
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/fieldAlertHelper.hpp:49]: Can this be a static assert?

**Review by sebasfalcone — COMMENTED** [review]
## Tests
I've added a CentOS 8 agent and checked the indexed vulnerabilities for this cases

```json
      {
        "_index": "wazuh-states-vulnerabilities-thinkpad-sebas",
        "_id": "001_55c76af225bcb6340ea4fd0ba8005e8c7284df69_CVE-2022-48648",
        "_score": 1,
        "_source": {
          "agent": {
            "id": "001",
            "name": "centos8",
            "type": "wazuh",
            "version": "v4.9.0"
          },
          "host": {
            "os": {
              "full": "CentOS Linux 8.5.2111",
              "kernel": "4.18.0-348.7.1.el8_5.x86_64",
              "name": "CentOS Linux",
              "platform": "centos",
              "type": "centos",
              "version": "8.5.2111"
            }
          },
          "package": {
            "architecture": "x86_64",
            "description": "The Linux kernel, based on version 4.18.0, heavily modified with backports",
            "installed": "2023-11-01T20:38:59.000Z",
            "name": "kernel",
            "size": 0,
            "type": "rpm",
            "version": "4.18.0-348.7.1.el8_5"
          },
          "vulnerability": {
            "category": "Packages",
            "classification": "-",
            "description": "In the Linux kernel, the following vulnerability has been resolved:  sfc: fix null pointer dereference in efx_hard_start_xmit  Trying to get the channel from the tx_queue variable here is wrong because we can only be here if tx_queue is NULL, so we shouldn't dereference it. As the above comment in the code says, this is very unlikely to happen, but it's wrong anyway so let's fix it.  I hit this issue because of a different bug that caused tx_queue to be NULL. If that happens, this is the error message that we get here:   BUG: unable to handle kernel NULL pointer dereference at 0000000000000020   [...]   RIP: 0010:efx_hard_start_xmit+0x153/0x170 [sfc]",
            "detected_at": "2024-09-16T16:37:02.580Z",
            "enumeration": "CVE",
            "id": "CVE-2022-48648",
            "published_at": "2024-04-28T13:15:07Z",
            "reference": "https://git.kernel.org/stable/c/0a242eb2913a4aa3d6fbdb86559f27628e9466f3, https://git.kernel.org/stable/c/8547c7bfc0617e7184e4da65b9b96681fcfe9998, https://git.kernel.org/stable/c/b3b41d4d95d3822b2e459ecbc80d030ea6aec5e7, https://git.kernel.org/stable/c/b3b952168ee1f220ba729fa100fd9d5aa752eb03",
            "scanner": {
              "vendor": "Wazuh"
            },
            "score": {
              "base": -1,
              "version": ""
            },
            "severity": "-",
            "under_evaluation": true
          },
          "wazuh": {
            "cluster": {
              "name": "Thinkpad-Sebas"
            },
            "schema": {
              "version": "1.0.0"
            }
          }
        }
      },
```

See that expected fields are filled with expected default values
```json
            "score": {
              "base": -1,
              "version": ""
            },
            "severity": "-",
            "under_evaluation": true
```

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/fieldAlertHelper.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+     * @param json      The JSON object to populate.
+     * @param key       The JSON pointer specifying the field to populate.
+     * @param value     The value to populate the field with.
+     *
+     * @note This function optimizes handling based on the type of the input value:
+     *  - If the value is a string (or string_view), the field is populated only if the string is non-empty after
+     * trimming.
+     *  - If the value is a floating-point number, the field is populated only if the value is not near zero (within a
+     * small epsilon range).
+     *  - If the value is an integral type, the field is populated only if the value is non-negative.
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/fieldAlertHelper.hpp:32]: This is not implemented and it should

Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/fieldAlertHelper.hpp]: Done!

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedAlertDetailsBuilder.hpp [hunk 3]
```diff
@@ -87,12 +88,22 @@ class TCVESolvedAlertDetailsBuilder final : public AbstractHandler<std::shared_p
...
                     {
                         json["vulnerability"]["cvss"][scoreVersion]["base_score"] =
-                            Utils::floatToDoubleRound(returnData.data->scoreBase(), 2);
+                            FieldAlertHelper::fillEmptyOrNegative(
+                                Utils::floatToDoubleRound(returnData.data->scoreBase(), 2));
                     }
                     json["vulnerability"]["enumeration"] = "CVE";
+                    json["vulnerability"]["cwe_reference"] = "-";
                     json["vulnerability"]["published"] = returnData.data->datePublished()->str();
+                    json["vulnerability"]["rationale"] = "-";
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedAlertDetailsBuilder.hpp:97]: Revert
```suggestion
```

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedAlertDetailsBuilder.hpp [hunk 4]
```diff
@@ -87,12 +88,22 @@ class TCVESolvedAlertDetailsBuilder final : public AbstractHandler<std::shared_p
                     if (!cvssVersion.empty())
                     {
                         json["vulnerability"]["cvss"][scoreVersion]["base_score"] =
-                            Utils::floatToDoubleRound(returnData.data->scoreBase(), 2);
+                            FieldAlertHelper::fillEmptyOrNegative(
+                                Utils::floatToDoubleRound(returnData.data->scoreBase(), 2));
                     }
                     json["vulnerability"]["enumeration"] = "CVE";
+                    json["vulnerability"]["cwe_reference"] = "-";
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedAlertDetailsBuilder.hpp:95]: Revert
```suggestion
```

**Review by sebasfalcone — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedAlertDetailsBuilder.hpp [hunk 5]
```diff
@@ -87,12 +88,20 @@ class TCVESolvedAlertDetailsBuilder final : public AbstractHandler<std::shared_p
                     if (!cvssVersion.empty())
                     {
                         json["vulnerability"]["cvss"][scoreVersion]["base_score"] =
-                            Utils::floatToDoubleRound(returnData.data->scoreBase(), 2);
+                            FieldAlertHelper::fillEmptyOrNegative(
+                                Utils::floatToDoubleRound(returnData.data->scoreBase(), 2));
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedAlertDetailsBuilder.hpp:92]: This field is not in the ones requested

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventPackageAlertDetailsBuilder.hpp [hunk 6]
```diff
@@ -142,15 +143,23 @@ class TEventPackageAlertDetailsBuilder final : public AbstractHandler<std::share
                     if (!cvssVersion.empty())
                     {
                         json["vulnerability"]["cvss"][scoreVersion]["base_score"] =
-                            Utils::floatToDoubleRound(returnData.data->scoreBase(), 2);
+                            FieldAlertHelper::fillEmptyOrNegative(
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventPackageAlertDetailsBuilder.hpp:146]: Ditto

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanOsAlertDetailsBuilder.hpp [hunk 7]
```diff
@@ -140,15 +141,23 @@ class TScanOsAlertDetailsBuilder final : public AbstractHandler<std::shared_ptr<
                     if (!cvssVersion.empty())
                     {
                         json["vulnerability"]["cvss"][scoreVersion]["base_score"] =
-                            Utils::floatToDoubleRound(returnData.data->scoreBase(), 2);
+                            FieldAlertHelper::fillEmptyOrNegative(
+                                Utils::floatToDoubleRound(returnData.data->scoreBase(), 2));
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanOsAlertDetailsBuilder.hpp:145]:
```suggestion
                            Utils::floatToDoubleRound(returnData.data->scoreBase(), 2);
```

**Review by sebasfalcone — APPROVED** [review]
LGTM!

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/fieldAlertHelper.hpp [hunk 8]
```diff
@@ -0,0 +1,66 @@
+#ifndef FIEL_ALERT_HELPER_HPP
+#define FIEL_ALERT_HELPER_HPP
+
+#include "numericHelper.h"
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/fieldAlertHelper.hpp]: unused

Comment (Dwordcito) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/fieldAlertHelper.hpp]:
```suggestion
```

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**